### PR TITLE
Added extensions to integrate ByteString with Kotlin stdlib APIs

### LINKED ISSUE #149: Add extensions to integrate ByteString with Kotlin stdlib APIs
Some Kotlin standard library APIs look natural to be integrated with ByteStings:
- Base64 (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io.encoding/-base64/)
- HexFormat (https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/text/HexFormat.kt)

The kx-io library should provide extensions facilitating use of ByteString with these APIs.

### PR DESCRIPTION
**Pull request by shanshin**
Implemented integration standard library APIs with ByteStings:
 - Base64
 - HexFormat

 Resolves #149

### CODE REVIEW

File: bytestring/common/src/Hex.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * Formats bytes in this byte string using the specified [format].
+ *
+ * Note that only [HexFormat.upperCase] and [HexFormat.BytesHexFormat] affect formatting.
+ *
+ * @param format the [HexFormat] to use for formatting, [HexFormat.Default] by default.
+ *
+ * @throws IllegalArgumentException if the result length is more than [String] maximum capacity.
+ */
+@ExperimentalStdlibApi
+@SinceKotlin("1.9")
```
Comment (qwwdfsad) [inline bytestring/common/src/Hex.kt:20]: I'm not sure if this is applicable anywhere outside of the standard library. Ilya may know for sure

File: bytestring/common/test/ByteStringHexTest.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+@file:OptIn(ExperimentalStdlibApi::class)
+
+package kotlinx.io.bytestring
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ByteStringHexTest {
```
Comment (fzhinkin) [inline bytestring/common/test/ByteStringHexTest.kt:14]: Please add a few test cases covering negative scenarios (where a call should fail with an exception), as you did for b64.

File: bytestring/common/src/Hex.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+
+@file:OptIn(ExperimentalStdlibApi::class)
+
+package kotlinx.io.bytestring
+
+/**
+ * Formats bytes in this byte string using the specified [format].
+ *
+ * Note that only [HexFormat.upperCase] and [HexFormat.BytesHexFormat] affect formatting.
```
Comment (fzhinkin) [inline bytestring/common/src/Hex.kt:13]: Should it be `bytes` instead of `BytesHexFormat`?

File: bytestring/common/src/Hex.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Formats bytes in this byte string using the specified [HexFormat].
+ *
+ * Note that only [HexFormat.upperCase] and [HexFormat.BytesHexFormat] affect formatting.
+ *
+ * @param startIndex the beginning (inclusive) of the subrange to format, 0 by default.
+ * @param endIndex the end (exclusive) of the subrange to format, size of this byte string by default.
+ * @param format the [HexFormat] to use for formatting, [HexFormat.Default] by default.
+ *
+ * @throws IndexOutOfBoundsException when [startIndex] or [endIndex] is out of range of this array indices.
```
Comment (fzhinkin) [inline bytestring/common/src/Hex.kt:34]: > this array indices

`this byte string indices`

**Review by fzhinkin — CHANGES_REQUESTED** [review]

File: bytestring/common/src/Hex.kt (hunk 3 above)
Comment (shanshin) [reply bytestring/common/src/Hex.kt:13]: I'm not sure. I did so for consistency with the stdlib docs

**Review by fzhinkin — APPROVED** [review]
